From c23e166a75172b14abf3aaf2b6801c55fd423f91 Mon Sep 17 00:00:00 2001
From: Martin Babinsky <mbabinsk@redhat.com>
Date: Tue, 16 Jun 2015 13:20:15 +0200
Subject: [PATCH] add DS index for userCertificate attribute

'eq' and 'pres' indices for userCertificate attribute allow for more efficient
lookup and matching of binary certificates assigned to users, hosts, and
services.

Part of http://www.freeipa.org/page/V4/User_Certificates

Reviewed-By: Martin Basti <mbasti@redhat.com>
---
 install/share/indices.ldif        | 9 +++++++++
 install/updates/20-indices.update | 7 +++++++
 2 files changed, 16 insertions(+)

diff --git a/install/share/indices.ldif b/install/share/indices.ldif
index 4f5bbf96e2adaec806954dfdf4d3f6c6c2286620..125888acbdbc1a7c7996113b250e8726034f1496 100644
--- a/install/share/indices.ldif
+++ b/install/share/indices.ldif
@@ -219,3 +219,12 @@ ObjectClass: top
 ObjectClass: nsIndex
 nsSystemIndex: false
 nsIndexType: eq
+
+dn: cn=userCertificate,cn=index,cn=userRoot,cn=ldbm database,cn=plugins,cn=config
+changetype: add
+cn: userCertificate
+ObjectClass: top
+ObjectClass: nsIndex
+nsSystemIndex: false
+nsIndexType: eq
+nsIndexType: pres
diff --git a/install/updates/20-indices.update b/install/updates/20-indices.update
index 405938139a7ff28018d9fc511ae7f22627be0b4a..a341e95f6cff5d9f7556fa1cda9a4e6036b3478c 100644
--- a/install/updates/20-indices.update
+++ b/install/updates/20-indices.update
@@ -129,3 +129,10 @@ default:ObjectClass: top
 default:ObjectClass: nsIndex
 default:nsSystemIndex: false
 default:nsIndexType: eq
+
+dn: cn=userCertificate,cn=index,cn=userRoot,cn=ldbm database,cn=plugins,cn=config
+default:cn: userCertificate
+default:ObjectClass: top
+default:ObjectClass: nsIndex
+only:nsSystemIndex: false
+only:nsIndexType: eq,pres
-- 
2.4.3

